packages/core-flows/src/definition/cart/workflows/add-shipping-method-to-cart.ts

```diff
@@ -24,22 +26,38 @@ export const addShippingMethodToWorkflow = createWorkflow(
   (
     input: WorkflowData<AddShippingMethodToCartWorkflowInput>
   ): WorkflowData<void> => {
+    const cart = useRemoteQueryStep({
+      entry_point: "cart",
+      fields: [
+        "region_id",
+        "currency_code",
+        "items.*",
+        "shipping_address.*",
+        "shipping_methods.*",
+      ],
+      variables: { id: input.cart_id },
+      list: false,
+    })
+
     const optionIds = transform({ input }, (data) => {
       return (data.input.options ?? []).map((i) => i.id)
     })
 
+    validateCartShippingOptionsStep({
```

Comment:
question: I wonder if we should just remove the method, if it's no longer applicable? We've talked about doing that previously. 

That would be a slightly better DX than telling users to handle this themselves in the client.

Reply:
Maybe it could be a config passed to the step?

Reply:
I created a new step called `refreshCartShippingMethodsStep` to do this, feels cleaner that way as the inputs are slightly different:

refreshCartShippingMethodsStep({ cart }) - removes shipping methods if no longer applicable
validateCartShippingOptionsStep({ cart, option_ids }) - throws error

should we merge them?

Reply:
I am fine to keep the two separated. In that case, I would probably unify the validation logic in a helper we can use in both methods, as this is identical right?

Reply:
I had a similar concern, maybe `refresh` can just return if a refresh was performed or not, and that can serve for the cases when we do want to validate? Just something that seemed confusing, but if you can't think of anything better than this is fine I guess

Reply:
The inputs are different and the outcome is different. They serve 2 different purposes even though they're doing similar things. If you both feel strongly enough, I'll merge them in.